sunpy/net/helioviewer.py

```diff
@@ -57,6 +57,14 @@ def data_sources(self):
         # Sort the output for printing purposes
         return OrderedDict(sorted(data_sources_dict.items(), key=lambda x: x[1]))
 
+    def _partial_match(self, key, dictionary):
+        """
+        A function that returns source id based on partial key.
+        """
+        for k, v in dictionary.items():
+            if all(k1 == k2 or k2 is None  for k1, k2 in zip(k, key)):
+                yield v
```

Comment:
Actually, this probably should be in a `util` file somewhere. It is too general to be a client function.

Reply:
So shall I remove this function from `helioviewer.py` and add it to `util`?

Reply:
Yeah, not sure what file to add it to but I think it belongs there.

Reply:
We can add it in `util.py` in the `util` folder

Reply:
Yeah that would work. Also the another test would need to be added to the util folder that works on a generic dictionary.

Reply:
Ok sure I will do that.